app/src/main/java/org/listenbrainz/android/presentation/features/yim/navigation/YimNavigation.kt

```diff
@@ -0,0 +1,36 @@
+package org.listenbrainz.android.presentation.features.yim.navigation
+
+import android.os.Build
+import androidx.annotation.RequiresApi
+import androidx.compose.runtime.Composable
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import org.listenbrainz.android.presentation.features.yim.YearInMusicActivity
+import org.listenbrainz.android.presentation.features.yim.YimViewModel
+import org.listenbrainz.android.presentation.features.yim.screens.YimHomeScreen
+import org.listenbrainz.android.presentation.features.yim.screens.YimMainScreen
+
+@RequiresApi(Build.VERSION_CODES.N)
```

Comment:
Avoid this at all times. We shouldn't be in favor of this while releasing a feature. Either we bump the minimum Android version to 24 or we look for alternatives. For now, I think we can bump the minimum Android version to 24

Reply:
I had an alternative but this makes life so much easier for all features. Internet observer has been added to util with proper documentation.